lib/src/component_declaration/component_base.dart

```diff
@@ -636,30 +636,49 @@ abstract class UiProps extends MapBase
+    List childArguments;
+    // Use `identical` since it compiles down to `===` in dart2js instead of calling equality helper functions,
+    // and we don't want to allow any object overriding `operator==` to claim it's equal to `_notSpecified`.
+    if (identical(c1, _notSpecified)) {
+      childArguments = [];
+    } else if (identical(c2, _notSpecified)) {
+      childArguments = [c1];
+    } else if (identical(c3, _notSpecified)) {
+      childArguments = [c1, c2];
+    } else if (identical(c4, _notSpecified)) {
+      childArguments = [c1, c2, c3];
+    } else if (identical(c5, _notSpecified)) {
+      childArguments = [c1, c2, c3, c4];
+    } else if (identical(c6, _notSpecified)) {
+      childArguments = [c1, c2, c3, c4, c5];
+    } else if (identical(c7, _notSpecified)) {
+      childArguments = [c1, c2, c3, c4, c5, c6];
+    } else {
+      childArguments = [c1, c2, c3, c4, c5, c6, c7, c8, c9, c10, c11, c12, c13, c14, c15, c16, c17, c18, c19, c20, c21, c22, c23, c24, c25, c26, c27, c28, c29, c30, c31, c32, c33, c34, c35, c36, c37, c38, c39, c40]
+        .takeWhile((child) => !identical(child, _notSpecified))
+        .toList();
+    }
 
-  /// Supports variadic children of the form `call([child1, child2, child3...])`.
-  @override
-  dynamic noSuchMethod(Invocation invocation) {
-    if (invocation.memberName == #call && invocation.isMethod) {
-      final positionalArguments = invocation.positionalArguments;
-      assert(_validateChildren(positionalArguments.length == 1 ? positionalArguments.single : positionalArguments));
-
-      final factory = componentFactory;
-      if (factory is ReactComponentFactoryProxy) {
-        // Use `build` instead of using emulated function behavior to work around DDC issue
-        // https://github.com/dart-lang/sdk/issues/29904
-        // Should have the benefit of better performance; TODO optimize type check?
-        // ignore: avoid_as
-        return factory.build(props, invocation.positionalArguments);
-      } else {
-        var parameters = []
-          ..add(props)
-          ..addAll(invocation.positionalArguments);
-        return Function.apply(factory, parameters);
-      }
+    assert(_validateChildren(childArguments.length == 1 ? childArguments.single : childArguments));
+
+    final factory = componentFactory;
+    if (factory is ReactComponentFactoryProxy) {
+      // Use `build` instead of using emulated function behavior to work around DDC issue
+      // https://github.com/dart-lang/sdk/issues/29904
+      // Should have the benefit of better performance; TODO optimize type check?
+      return factory.build(props, childArguments);
+    } else {
+      var parameters = []
+        ..add(props)
+        ..addAll(childArguments);
+      return Function.apply(factory, parameters);
     }
+  }
 
+  @override
+  dynamic noSuchMethod(Invocation invocation) {
```

Comment:
@aaronlademann-wf @greglittlefield-wf I wanted to delete this method from this class, since it seems superfluous since it's just invoking `noSuchMethod` on `Object`. However, deleting it caused analyzer (and build errors in dart 2) such as: 
<img width="1188" alt="screen shot 2018-12-07 at 3 37 45 pm" src="https://user-images.githubusercontent.com/12497362/49677131-4061d280-fa3a-11e8-86a4-71ee5b3a370a.png">
Those analyzer errors appear in dart 1 and dart 2 if this NSM implementation is removed. I'm not sure why, and that makes me uncomfortable. Any thoughts on why this might be?

Reply:
Oh weird... I think there might be support in the analyzer to assume an object implements methods it actually doesn't if `noSuchMethod` is missing. I thought that was a call-time thing, though... 

Either way, if removing this is causing issues, then there may be some other underlying problems either:
- in the dartanalyzer
- in our props class hierarchies (I think this is more likely)

Reply:
Turns out this was an issue with several class hierarchies in WSD. Fixed in https://github.com/Workiva/web_skin_dart/pull/1100